arbor/include/arbor/morph/locset.hpp

```diff
@@ -138,4 +141,9 @@ locset nil();
 
 } // namespace ls
 
+// union of two locsets
+locset join(locset, locset);
+// multiset sum of two locsets
+locset sum(locset, locset);
+
```

Comment:
I reckon it'd be cleaner to put lambdas in the S-expression parser function table, than to add these extra constructors and functions: it saves doubling up on the interfaces.

Reply:
The functions that you have hilighted here are not actually new calls.
I added their prototypes here so that the can be referred to (the only place their prototype appeared before was as `friend` calls inside the `locset`/`region` class definition.)

I think it is cleaner to directly refer to an existing function if it exists, and restrict use of lambdas to situations where no calls exist.

I have, however, removed the `location(msize_t, double)` prototype and its implementation, and replaced it with a lambda in the s-expresson table.